packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -60,7 +60,7 @@ export const isValidCodeWhispererCoreConnection = (conn?: Connection): conn is C
     }
 
     if (isSageMaker()) {
-        return isIamConnection(conn)
+        return isIamConnection(conn) || (isSsoConnection(conn) && hasScopes(conn, codeWhispererCoreScopes))
```

Comment:
Can we use `amazonQScopes`? Does sagemaker SSO auth allow for the complete Q scope set?
```suggestion
        return isIamConnection(conn) || (isSsoConnection(conn) && hasScopes(conn, amazonQScopes))
```
It is code smell that we have these scope subsets, so it is easy to confuse which one needs to be used/checked. We have plans to fix this.

Reply:
No it doesn't, but tackling the scopes is a separate issue that I'll address later on. For now, if user login with their SSO credentials, they will go through the normal login flow which appends the full set of credentials for them. I can omit the scope check from here though as it doesn't make a difference for Sagemaker users.

Reply:
For now, I'll simplify the condition and put in a separate function for readability. it ignores any scope checking as we're handling deactivating features not supported by sagemaker in a separate change